darts/models/forecasting/torch_forecasting_model.py

```diff
@@ -916,6 +916,167 @@ def fit_from_dataset(
         self._train(train_loader, val_loader)
         return self
 
+    @staticmethod
+    def setup_finetuning(
```

Comment:
How about extending `load_from_checkpoint()` instead? By making it prepare the model for further training when the extra parameters are passed?
If we keep it separate, I'd rename this one to something like `load_for_training()`. To make it clear that (i) it's loading a model, and (ii) this is for further training ("finetuning" is a little bit more specialised than this and could be understood as training some layers only)

Reply:
Extending `load_from_checkpoint()` should be possible, I created another method because I wanted them to cover a different scenario: `load_from_checkpoint()` for when the training have to be resumed (no parameter change) and `setup_finetuning()` if the training is explicitly performed for additional steps, eventually with another scheduler/optimizer/trainer parameters.
I agree with you, the name is a bit confusing, I will adapt it.